<a href="https://colab.research.google.com/github/br1lamasdiaz/hacking-civico/blob/master/tareas/02-python-101.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Densidad 

In [84]:
directorio = [ {"ciudad": "colima_municipio","poblacion": 150673,"territorio": 1668},
               {"ciudad": "colima_estado","poblacion": 711235,"territorio": 5625},
               {"ciudad": "mexico","poblacion": 119938473,"territorio": 1953162},]


In [85]:
for miembro in directorio:
    print("Ubicacion:", miembro['ciudad'])
    print("Poblacion:", miembro['poblacion'])
    print("Territorio:", miembro['territorio'])
    print()

Ubicacion: colima_municipio
Poblacion: 150673
Territorio: 1668

Ubicacion: colima_estado
Poblacion: 711235
Territorio: 5625

Ubicacion: mexico
Poblacion: 119938473
Territorio: 1953162



In [86]:
def densidad (poblacion, territorio):
  return (poblacion/territorio)

In [87]:
for t in directorio:
  densidad_poblacion = densidad (t['poblacion'], t['territorio'])
  print('La densidad poblacional es ' + str(densidad_poblacion) + ' en ' + t['ciudad'])

La densidad poblacional es 90.33153477218225 en colima_municipio
La densidad poblacional es 126.44177777777777 en colima_estado
La densidad poblacional es 61.40733487544812 en mexico


La **densidad de población** es una medida de distribución de población de un país, estado o municipio que es equivalente al número de habitantes dividido entre el área del territorio.
 
| Ubicacion | Poblacion   | Territorio | Densidad |
| --------- | ----------- | ---------- | -------- |
|  M. Col   |    150673   |    1668    | 90.3     |
|  E. Col   |    711235   |    5625    | 126.4    |
|  Mexico   | 119938473   | 1953162    | 61.4     |





# Inversión

In [123]:
datos = [ {"año":"2017", "gasto":602799942, "poblacion":123597},
          {"año":"2018", "gasto": 640608041, "poblacion":150673},]

In [127]:
for inversion in datos:
  print("En", inversion["año"], "se invirtió la cantidad de $", inversion["gasto"])

En 2017 se invirtió la cantidad de $ 602799942
En 2018 se invirtió la cantidad de $ 640608041


In [130]:
def percapita (gasto, poblacion):
   return (gasto/poblacion)

In [134]:
for i in datos:
  gasto_percapita = percapita (i["gasto"], i["poblacion"])
  print('La inversion per capita en ' + i["año"] + "fue de $ " + str(gasto_percapita))

La inversion per capita en 2017fue de $ 4877.140561664118
La inversion per capita en 2018fue de $ 4251.64456140118


**¿Para qué crees que nos pueda ser útil esta información?**

En caso de la densidad de población y la inversión per capita, nos ayuda a dimensionar de forma más exacta las condiciones de vida en determinadas ciudades, ya que nos permite conocer un dato mas apegado a la realidad por persona, y no por comunidad. 

**¿Qué decisiones podría tomar el gobierno basado en ésta?**

El nivel de gasto o territorio per capita permite dimensionar la condicion e ingerencia que puede tener las instituciones públicas en la sociedad, quiza un territorio tenga un menor gasto si se copara el dato duro junto con otros presupuestos, pero no quiere decir que el sistema provea de una mejor atencion y capacidad de gasto por usuario.
